homeassistant/components/utility_meter/sensor.py

```diff
@@ -147,8 +157,21 @@ def __init__(
             self._name = f"{source_entity} meter"
         self._unit_of_measurement = None
         self._period = meter_type
-        self._period_offset = meter_offset
-        self._cron_pattern = cron_pattern
+        if meter_type is not None:
+            # For backwards compatibility reasons we convert the period and offset into a cron pattern
+            if meter_offset.days >= 31:
+                _LOGGER.error(
+                    "Unsupported offset, please use a cron pattern. Continuing without offset"
+                )
+                meter_offset = timedelta()
+            self._cron_pattern = PERIOD2CRON[meter_type].format(
+                minute=meter_offset.seconds % 3600 // 60,
+                hour=meter_offset.seconds // 3600,
+                day=meter_offset.days + 1,
+            )
+            _LOGGER.debug("CRON pattern: %s", self._cron_pattern)
```

Comment:
At least the error, and possible this entire block, should be moved to the schema.